src/Middleware/AuthorizationMiddleware.php

```diff
@@ -101,21 +104,44 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
         $attribute = $this->getConfig('identityAttribute');
         $identity = $request->getAttribute($attribute);
 
-        if ($identity === null) {
-            return $next($request, $response);
+        if ($identity !== null) {
+            $identity = $this->buildIdentity($service, $identity);
+            $request = $request->withAttribute($attribute, $identity);
         }
 
-        $identity = $this->buildIdentity($service, $identity);
-
-        $request = $request->withAttribute($attribute, $identity);
-        $response = $next($request, $response);
+        try {
+            $response = $next($request, $response);
+        } catch (Exception $exception) {
+            $handler = $this->getHandler();
+            $response = $handler->handle($exception, $request, $response, (array)$this->getConfig('unauthorizedHandler'));
```

Comment:
Should options be passed into the handler constructor? That would allow the objects to validate options earlier.